cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -328,7 +334,8 @@ class aggregate_reader_metadata {
  public:
   aggregate_reader_metadata(host_span<std::unique_ptr<datasource> const> sources,
                             bool use_arrow_schema,
-                            bool has_cols_from_mismatched_srcs);
+                            bool has_cols_from_mismatched_srcs,
+                            bool read_page_indexes = true);
```

Comment:
Not too happy with adding another bool parameter here, but using an enum makes a few things a lot more verbose. Would prefer to systematically change all bools here to enums in a separate. Open to suggestions.

Reply:
I think this is okay as is for now